lms/templates/edxnotes/note-item.underscore

```diff
@@ -17,7 +17,12 @@
     <ol class="note-comments" role="region" aria-label="<%- gettext('Note') %>">
         <li class="note-comment">
             <p class="note-comment-title"><%- gettext("You commented...") %></p>
-            <p class="note-comment-p"><%= text %></p>
+            <p class="note-comment-p">
+                <%= interpolate_text(_.escape(text), {
+                    highlighted_notes_start: '<span class="note-highlight">',
```

Comment:
The note is stored in the database with no highlighting. When a user performs a search,  elasticsearch and our Python code are adding {highlighted_notes_start} and a matching end tag around matches for the search term. 

Let's say we wanted this tag replacement to happen in underscore, and not mako, and we were using the new ${tag_name} interpolation.  Then this would need to be ${highlighted_notes_start} that gets inserted by Python.  So, the python code needs to know the proper delims to use.

Given that, as you both point out, the alternative puts the <span> in Python, which I'm happy to concede might be the worse evil.

Given you both agree, and there's no perfect answer, I'm happy to go along.

> On Jan 14, 2016, at 5:40 PM, Andy Armstrong <notifications@github.com> wrote:
> 
> In lms/templates/edxnotes/note-item.underscore:
> 
> > @@ -17,7 +17,12 @@
> >      <ol class="note-comments" role="region" aria-label="<%- gettext('Note') %>">
> >          <li class="note-comment">
> >              <p class="note-comment-title"><%- gettext("You commented...") %></p>
> > -            <p class="note-comment-p"><%= text %></p>
> > +            <p class="note-comment-p">
> > +                <%= interpolate_text(_.escape(text), {
> > +                    highlighted_notes_start: '<span class="note-highlight">',
> My preference is for option 1 because the JavaScript doesn't have to trust that the server performed the necessary escaping. As @muzaffaryousaf says, the code then performs the escaping as late as possible. It also means that the JS has access to the raw text which might be useful in other situations where escaping isn't necessary (e.g. setting the value through JQuery).
> 
> @robrap I'm not sure what you mean by "the tag definition (name + delimiters) is in Python". If I'm reading this correctly, the text is the raw bookmark text as stored in the database, and then everything else lives in the Underscore file.
> 
> —
> Reply to this email directly or view it on GitHub.
>